src/lib.rs

```diff
@@ -686,6 +1070,53 @@ impl TestData {
 mod tests {
     use super::*;
 
+    #[cfg(all(test, feature = "alloc", feature = "kem"))]
+    #[test]
+    fn test_katkem() {
+        use std::env;
+
+        std::thread::Builder::new()
+            .stack_size(10 * 1024 * 1024)
+            .spawn(|| {
```

Comment:
concerning the thread stack size increase, I attempted to write a small wrapper function using generics in order to just wrap the three KEM function calls and not the other code, however could not get it to work out, maybe you have some suggestions to fix that:
```rust
fn increase_stack_size_wrapper<F: Fn() -> R + Send>(function: F) {
    std::thread::Builder::new()
        .stack_size(10 * 1024 * 1024)
        .spawn(|| { 
            function
        })
        .unwrap()
        .join()
        .unwrap();
}
```